contents/java/Spring.md

```diff
@@ -0,0 +1,193 @@
+```java
+    package com.company;  
+  
+    public class Employee {  
+       private int id;  
+       private String name;  
+       private String city;  
+  
+       public int getId() {  
+           return id;  
+       }  
+       public void setId(int id) {  
+           this.id = id;  
+       }  
+       public String getName() {  
+           return name;  
+       }  
+       public void setName(String name) {  
+           this.name = name;  
+       }  
+       void display(){  
+           System.out.println(id+" "+name);  
+       }
+    }
+```
+    
+And the corresponding `applicationContext.xml` file could be:
+```
+    <?xml version="1.0" encoding="UTF-8"?>  
+    <beans  
+    xmlns="http://www.springframework.org/schema/beans"  
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"  
+    xmlns:p="http://www.springframework.org/schema/p"  
+    xsi:schemaLocation="http://www.springframework.org/schema/beans  
+                http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">  
+  
+    <bean id="obj" class="com.company.Employee">  
+    <property name="id">  
+    <value>20</value>  
+    </property>  
+    <property name="name">  
+    <value>Amy</value>  
+    </property>  
+    </bean>  
+    </beans>
+```
+We are providing the information into the bean by this file. The property element invokes the setter method. The value subelement of property will assign the specified value.
+   
+To better understand how this dependency injection work, we can have a `Test.java` looks like the following:
+```java
+    public class Test {  
+        public static void main(String[] args) {   
+            Resource r=new ClassPathResource("applicationContext.xml");  
+            BeanFactory factory=new XmlBeanFactory(r);  
+          
+            Employee e=(Employee)factory.getBean("obj");  
+            s.display();
+        }  
+    }  
+```
+The out put of the above code will be `20 Amy`.
+
+### Spring JdbcTemplate
+
+Spring JdbcTemplate is a powerful mechanism to connect to the database and execute SQL queries. It internally uses
+JDBC API, but eliminates a lot of problems of it.
+    
+The problems of traditional JDBC API are listed as follows:
+- We need to write a lot of code before and after executing the query, such as creating connection,
+statement, closing resultset, connection etc.
+- We need to perform exception handling code on the database logic.
+- We need to handle transaction.
+- Repetition of all these codes from one to another database logic is a time consuming task.
+
+Spring JdbcTemplate eliminates all the above mentioned problems of JDBC API.
+It provides you methods to write the queries directly, so it saves a lot of work and time.
+![Spring JdbcTemplate](springJDBC.png "How much Spring saves you")
+
+### Spring AOP
+
+Aspect Oriented Programming (AOP) compliments OOPs in the sense that it provides modularity in aspect instead of class.
```

Comment:
Consider rephrasing by introducing the cause before the effect for clarity.
Something like: 
`AOP is ... and has the effect of ... because ...`